engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1011,7 +1012,7 @@ private void execute() throws YarnException, IOException
             numCompletedContainers.incrementAndGet();
             LOG.info("Container completed successfully." + ", containerId=" + containerStatus.getContainerId());
             // Reset counter for node failure, if exists
-            String hostname = allocatedContainer.container.getNodeId().getHost();
+            String hostname = allocatedContainer.host;
```

Comment:
Can allocatedContainer be null, so we should check before de-referencing it? Ideally it should not when exitStatus == 0, but may we can prevent another NPE with a check?